src/pow.cpp

```diff
@@ -24,42 +24,38 @@ unsigned int GetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHead
     if (pindexLast == NULL)
         return nProofOfWorkLimit;
 
-    const CBlockIndex* pindexBits = pindexLast;
-    {
-        if (params.fPowAllowMinDifficultyBlocks)
-        {
-            // Special difficulty rule for testnet:
-            // If the new block's timestamp is more than 2* 2.5 minutes
-            // then allow mining of a min-difficulty block.
-            if (pblock->GetBlockTime() > pindexLast->GetBlockTime() + params.nPowTargetSpacing*2)
-                return nProofOfWorkLimit;
-            else {
-                // Get the last non-min-difficulty (or at worst the genesis difficulty)
-                while (pindexBits->pprev && pindexBits->nBits == nProofOfWorkLimit)
-                    pindexBits = pindexBits->pprev;
-            }
-        }
-    }
-
     // Find the first block in the averaging interval
     const CBlockIndex* pindexFirst = pindexLast;
+    arith_uint256 bnAvg;
+    arith_uint256 bnTmp;
     for (int i = 0; pindexFirst && i < params.nPowAveragingWindow; i++) {
+        if (i == 0) {
+            bnAvg.SetCompact(pindexFirst->nBits);
+        } else {
+            bnTmp.SetCompact(pindexFirst->nBits);
+            bnAvg = ((bnAvg * i) + bnTmp) / (i + 1);
+        }
         pindexFirst = pindexFirst->pprev;
     }
 
     // Check we have enough blocks
     if (pindexFirst == NULL)
         return nProofOfWorkLimit;
 
-    return CalculateNextWorkRequired(pindexBits->nBits, pindexLast->GetMedianTimePast(), pindexFirst->GetMedianTimePast(), params);
+    return CalculateNextWorkRequired(bnAvg, pindexLast->GetMedianTimePast(), pindexFirst->GetMedianTimePast(), params);
 }
 
+// Left for testing purposes
```

Comment:
Not the case; `GetDifficultyINTERNAL()` uses `GetNextWorkRequired()` like the rest of the code. The only user of this particular method is `test/pow_tests.cpp`.